### PR TITLE
test: async plugin errors throw at an indeterminate point, causing flake in crash reporting error

### PR DESCRIPTION
**Pull request by cacieprins**
- Closes 

### Additional details

This test can flake because the unhandled exception in the plugin happens at an indeterminate time: we cannot assert on which tests have completed or failed. This removes stdout snapshots for this test, and changes the request assertion to only check the error message that was reported.

Example: https://app.circleci.com/pipelines/github/cypress-io/cypress/59603/workflows/0be56496-3489-4645-a79c-8d28aa818d7c/jobs/2478708

### PR Tasks

- [X] Have tests been added/updated?
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: system-tests/test/record_spec.js [hunk 1]
```diff
@@ -2354,20 +2354,13 @@ describe('e2e record', () => {
...
-            snapshot: true,
+            snapshot: false,
             expectedExitCode: 1,
           }).then(() => {
             const requests = getRequests()
             const postResultsRequest = requests.find((r) => r.url === `POST /instances/${instanceId}/results`)
 
-            console.log(postResultsRequest)
             expect(postResultsRequest?.body.exception).to.include('Your configFile threw an error')
-            expect(postResultsRequest?.body.tests).to.have.length(2)
```
Comment (jennifer-shehane) [inline system-tests/test/record_spec.js:2365]: @cacieprins I'd suppose the original intent of this was just to make sure we didn't blow away the results when there's an error and that we exit properly. Seems like you could still keep the `body.tests - to have length 2`..... is that assertion really adding more value than not having it though? 🤔

**Review by jennifer-shehane — APPROVED** [review]
@cacieprins

File: system-tests/test/record_spec.js (hunk 1 above)
Comment (cacieprins) [reply system-tests/test/record_spec.js:2365]: Sometimes the before spec config listener can error before the reporter knows how many tests there are, so unfortunately even that is indeterminate for this kind of crash 

The primary purpose with this system test is to make sure we're sending an error report to the cloud, even if a plugin fails with an unhandled exception in an async manner

**Review by emilyrohrbough — COMMENTED** [review]
Any reason we don't test this with a spec that only has 1 test so it is deterministic? I imagine that spec could have a cy.pause() to ensure the error is observed?

**Review by emilyrohrbough — COMMENTED** [review]
Any reason we don't test this with a spec that only has 1 test so it is deterministic? I imagine that spec could have a cy.pause() to ensure the error is observed?

**Comment by cacieprins** [comment]
> Any reason we don't test this with a spec that only has 1 test so it is deterministic? I imagine that spec could have a cy.pause() to ensure the error is observed?

If the error is thrown soon enough in the cycle, `tests` will actually be null because they haven't been reported to the reporter yet

**Review by emilyrohrbough — APPROVED** [review]

**Pull request merged**